satellite/overlay/service.go

```diff
@@ -342,14 +315,11 @@ func (service *Service) FindStorageNodesWithPreferences(ctx context.Context, req
 		OnlineWindow:     preferences.OnlineWindow,
 		DistinctIP:       preferences.DistinctIP,
 	}
-	reputableNodes, err := service.db.SelectStorageNodes(ctx, reputableNodeCount-len(newNodes), &criteria)
+	nodes, err = service.db.SelectStorageNodes(ctx, reputableNodeCount, newNodeCount, &criteria)
```

Comment:
Let's add a test for SelectStorageNodes where `reputableNodeCount = N` and `newNodeCount = K` and there are `N + K` total nodes where all of them are reputable. In that case it should end up selecting all of the reputable nodes.

Reply:
And a case where "reputable node count = N" and "new node count = K" and ensure that it only ends up selecting MinimumRequiredNodes or RequestedCount. It feels like it won't return the same number of nodes as it did before.

Reply:
`reputableNodeCount` here is actually "required node count", how many nodes to return.

Reply:
For your first comment, this test exists already here, and the code passes this test: https://github.com/storj/storj/blob/master/satellite/overlay/selection_test.go#L198
I added TestEnsureMinimumRequested for the other concern.

Reply:
The `TestEnsureMinimumRequested` had few bugs in it, I fixed them and now it is failing, whereas with the old code it passes.